hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/util/SparkKeyGenUtils.scala

```diff
@@ -34,26 +34,26 @@ object SparkKeyGenUtils {
    * @return partition columns
    */
   def getPartitionColumns(props: TypedProperties): String = {
-    val keyGeneratorClass = getKeyGeneratorClassName(props)
-    getPartitionColumns(keyGeneratorClass, props)
+    val keyGenerator = HoodieSparkKeyGeneratorFactory.createKeyGenerator(props)
```

Comment:
we can't do this. For auto key geneator, the key gen has to be invoked from within executor. can we avoid instantiating key gen class here